tests/Agent/IntegrationTests/IntegrationTests/InfiniteTracing/InfiniteTracingFlakyTests.cs

```diff
@@ -0,0 +1,151 @@
+﻿// Copyright 2020 New Relic, Inc. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using MultiFunctionApplicationHelpers;
+using NewRelic.Agent.IntegrationTestHelpers;
+using NServiceBus.Features;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace NewRelic.Agent.IntegrationTests.InfiniteTracing
+{
+    public abstract class InfiniteTracingFlakyTestsBase<TFixture> : NewRelicIntegrationTest<TFixture>
+        where TFixture : ConsoleDynamicMethodFixture
+    {
+        private readonly TFixture _fixture;
+
+        public InfiniteTracingFlakyTestsBase(TFixture fixture, ITestOutputHelper output) : base(fixture)
+        {
+            _fixture = fixture;
+            _fixture.TestLogger = output;
+            _fixture.SetTimeout(System.TimeSpan.FromMinutes(5));
+
+            // set an 80% chance that the trace observer will throw an error
+            _fixture.RemoteApplication.SetAdditionalEnvironmentVariable("NEW_RELIC_INFINITE_TRACING_SPAN_EVENTS_TEST_FLAKY", "90");
+
+            // set the code to be returned when the trace observer throws an error
+            // must be a value from 0 to 16 as per https://github.com/grpc/grpc/blob/master/doc/statuscodes.md
+            // use the StatusCode enum from gRPC 
+            _fixture.RemoteApplication.SetAdditionalEnvironmentVariable("NEW_RELIC_INFINITE_TRACING_SPAN_EVENTS_TEST_FLAKY_CODE",
+                ((int)Grpc.Core.StatusCode.Internal).ToString());
+
+            //_fixture.AddCommand("RootCommands LaunchDebugger");
+
+            _fixture.AddCommand("InfiniteTracingTester StartAgent");
+
+            _fixture.AddCommand("RootCommands DelaySeconds 5"); // give the agent time to warm up
+
+            _fixture.AddCommand("InfiniteTracingTester Make8TSpan");
+            _fixture.AddCommand("InfiniteTracingTester Make8TSpan");
+            _fixture.AddCommand("InfiniteTracingTester Make8TSpan");
+            _fixture.AddCommand("InfiniteTracingTester Make8TSpan");
+            _fixture.AddCommand("InfiniteTracingTester Make8TSpan");
+            _fixture.AddCommand("InfiniteTracingTester Make8TSpan");
```

Comment:
90% error rate could still result in none of these having an error. We should set the error rate to 100% imho

Reply:
Set it to 100%